### PR TITLE
attention assert

### PR DESCRIPTION
**Pull request by hhexiy**
## Description ##
Luong style attention should make sure that the projected query has the same dimension as the key, otherwise batch_dot will have dimension mismatch.

### Changes ###
- add dimension assert for Luong style attention

### CODE REVIEW

File: gluonnlp/model/attention_cell.py [hunk 1]
```diff
@@ -457,8 +457,8 @@ def _compute_weight(self, F, query, key, mask=None):
             if not self._luong_style:
                 key = self._proj_key(key)
             else:
-                assert query.shape[-1] == key.shape[-1], 'Luong style attention requires key to have ' \
-                                                         'the same dim as the projected query'
+                assert query.shape[-1] == key.shape[-1], 'Luong style attention requires key to ' \
+                                                         'have the same dim as the projected query'
```
Comment (szha) [inline gluonnlp/model/attention_cell.py]: could you include the shapes in the error message?

File: gluonnlp/model/attention_cell.py [hunk 2]
```diff
@@ -456,6 +456,11 @@ def _compute_weight(self, F, query, key, mask=None):
             query = self._proj_query(query)
             if not self._luong_style:
                 key = self._proj_key(key)
+            else:
+                assert query.shape[-1] == key.shape[-1], 'Luong style attention requires key to ' \
```
Comment (szha) [inline gluonnlp/model/attention_cell.py]: query.shape is only available if query is ndarray. you can do the check by checking whether `F == nd`.

Comment (hhexiy) [reply gluonnlp/model/attention_cell.py]: added.

**Comment by mli** [comment]
Job PR-201/5 is complete. 
Docs are uploaded to http://gluon-nlp-staging.s3-accelerate.dualstack.amazonaws.com/PR-201/5/index.html

**Pull request merged**